components/toast.js

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useCallback, useContext, useState } from 'react'
+import Toast from 'react-bootstrap/Toast'
+import ToastBody from 'react-bootstrap/ToastBody'
+import ToastContainer from 'react-bootstrap/ToastContainer'
+import ToastHeader from 'react-bootstrap/ToastHeader'
+import { randInRange } from '../lib/rand'
+
+const ToastContext = createContext(() => {})
+
+export const ToastProvider = ({ children }) => {
+  const [toasts, setToasts] = useState([])
+  const dispatchToast = useCallback((toastConfig) => {
+    const id = Array(10).fill(randInRange(0, 9)).join('')
+    toastConfig = {
+      ...toastConfig,
+      id
+    }
+    if (toastConfig.autohide && toastConfig.delay > 0) {
+      setTimeout(() => {
+        setToasts(toasts => toasts.filter(toast => toast.id !== id))
+      }, toastConfig.delay)
+    }
+    setToasts(toasts => [...toasts, toastConfig])
+  }, [])
+  const removeToast = useCallback(id => {
+    setToasts(toasts => toasts.filter(toast => toast.id !== id))
+  }, [])
+  const getHeaderText = useCallback((toastConfig) => {
+    return toastConfig.header ?? {
+      success: 'Success',
+      warning: 'Warning',
+      danger: 'Danger',
+      info: 'Info'
+    }[toastConfig.variant] ?? toastConfig.variants
+  }, [])
+  return (
+    <ToastContext.Provider value={dispatchToast}>
+      <ToastContainer position='bottom-end' containerPosition='fixed'>
+        {toasts.map(toast => (
+          <Toast key={toast.id} bg={toast.variant} show autohide={false} onClose={() => removeToast(toast.id)}>
+            <ToastHeader closeButton closeLabel='close'><span className='flex-grow-1'>{getHeaderText(toast)}</span></ToastHeader>
```

Comment:
I am wondering if the header is required? Looks a bit unnecessary. The color already tells you if it's a success or error. Is it possible to put the close button into the body?

I like the styling of the [evergreen toaster](https://evergreen.segment.com/components/toaster) a lot more:


https://github.com/stackernews/stacker.news/assets/27162016/3365c607-6404-454a-9012-b54d6ebfd3b7

Reply:
I agree, it does feel like overkill to have the header when it just says 'success'. Let me explore some options here. I do like the evergreen styling more as well.

Reply:
I don't like the X button though. Wouldn't it be better if you could just click the toast anywhere to close it? Are there cases where I would click the toast which are not about closing it?

Just some ideas, you don't have to do everything I am saying haha

We can always iterate on features again later

Reply:
Generally speaking, I have seen toast-like messages that are "actionable", meaning clicking on them takes you somewhere within the app. This is where an explicit close button/icon helps users know that they can dismiss the toast w/o it launching them into some other flow.

I don't think we have such a use case right now, but something to consider, I suppose.

Reply:
10 out of 10 programmers think close buttons aren't necessary (most of our modals didn't have them to begin with) and 10 out of 10 UX designers think you should have them.

I think the UX designers are right. People hate overlays and seeing/knowing that they can close them, comforts them.

Reply:
To admit, I didn't meant to remove the X button but just to make the hitbox of the X button the whole toast